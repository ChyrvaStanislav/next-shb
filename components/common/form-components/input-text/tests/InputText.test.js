import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Formik } from 'formik';
import InputText from '../index';

const setFieldValue = jest.fn();
const onBlur = jest.fn();
const onChange = jest.fn();
const customChange = jest.fn();

const event = {
  target: {
    value: 'some value'
  }
};

const Component = props => (
  <Formik>
    <InputText
      field={{
        name: 'name',
        value: '',
        onBlur,
        onChange,
      }}
      form={{
        touched: {},
        errors: {},
        setFieldValue,
      }}
      type="text"
      placeholder="Some placeholder"
      formId="formId"
      label="Some label"
      classNameWrapper="someClass"
      customChange={customChange}
      {...props}
    />
  </Formik>
);

describe('InputText renders', () => {
  it('should InputText renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Events works', () => {
  it('should onBlur works', () => {
    const { getByTestId } = render(<Component />);
    const input = getByTestId('input-text');
    fireEvent.blur(input, event);

    expect(onBlur).toBeCalled();
    expect(setFieldValue).toBeCalled();
    expect(input.value).toBe('some value');
  });

  it('should onChange works', () => {
    const { getByTestId } = render(<Component />);
    const input = getByTestId('input-text');
    fireEvent.change(input, event);

    expect(onChange).toBeCalled();
    expect(customChange).toBeCalled();
  });
});


describe('Error display', () => {
  it('should Error message displays', () => {
    const { getByText } = render(<Component form={{
      touched: {
        name: true,
      },
      errors: {
        name: 'Some error'
      },
    }}
    />);

    const errorIcon = getByText('Some error');
    expect(errorIcon).toBeInTheDocument();
  });
});
