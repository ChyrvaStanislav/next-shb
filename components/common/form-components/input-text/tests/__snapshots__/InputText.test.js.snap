// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText renders should InputText renders 1`] = `
<DocumentFragment>
  <section
    class="textareaContainer"
  >
    <label
      class="small_filters label"
      for="formId-name"
      other=""
    >
      Some label
    </label>
    <div
      class="inputWrapper someClass"
    >
      <input
        autocomplete="off"
        class="input"
        data-testid="input-text"
        id="formId-name"
        name="name"
        placeholder="Some placeholder"
        type="text"
        value=""
      />
    </div>
  </section>
</DocumentFragment>
`;
